File: tools/python-execute-eval/solutions/notes-crud.mjs
import { kody } from 'kody:runtime'

export default async function main(params) {
	for (const operation of params.ops) {
		switch (operation.op) {
			case 'write':
				await kody.call('notes.write', {
					id: operation.id,
					text: operation.text,
				})
				break
			case 'remove':
				await kody.call('notes.remove', { id: operation.id })
				break
			default:
				throw new Error(`Unknown operation: ${operation.op}`)
		}
	}
	return kody.call('notes.list', {})
}

File: tools/python-execute-eval/solutions/notes-crud.py
async def main(params):
    for operation in params["ops"]:
        if operation["op"] == "write":
            await kody.call("notes.write", {
                "id": operation["id"],
                "text": operation["text"],
            })
        elif operation["op"] == "remove":
            await kody.call("notes.remove", {"id": operation["id"]})
    return await kody.call("notes.list", {})
